### ISSUE TITLE
Reading custom extensions

### ISSUE THREAD

**Issue by dainnilsson** [pyca/cryptography#2288]
I need to be able to read the value of custom X509 extensions from a certificate. Currently, after loading the certificate using x509.load_pem_x509_certificate(), the extensions object is empty. As a reference, I want to be able to read out the `1.3.6.1.4.1.45724.2.1.1` extension from the following certificate:

```
-----BEGIN CERTIFICATE-----
MIICIjCCAQygAwIBAgIEIHHwozALBgkqhkiG9w0BAQswDzENMAsGA1UEAxMEdGVz
dDAeFw0xNTA4MTEwOTAwMzNaFw0xNjA4MTAwOTAwMzNaMCkxJzAlBgNVBAMTHll1
YmljbyBVMkYgRUUgU2VyaWFsIDU0NDMzODA4MzBZMBMGByqGSM49AgEGCCqGSM49
AwEHA0IABPdFG1pBjBBQVhLrD39Qg1vKjuR2kRdBZnwLI/zgzztQpf4ffpkrkB/3
E0TXj5zg8gN9sgMkX48geBe+tBEpvMmjOzA5MCIGCSsGAQQBgsQKAgQVMS4zLjYu
MS40LjEuNDE0ODIuMS4yMBMGCysGAQQBguUcAgEBBAQDAgQwMAsGCSqGSIb3DQEB
CwOCAQEAb3YpnmHHduNuWEXlLqlnww9034ZeZaojhPAYSLR8d5NPk9gc0hkjQKmI
aaBM7DsaHbcHMKpXoMGTQSC++NCZTcKvZ0Lt12mp5HRnM1NNBPol8Hte5fLmvW4t
Q9EzLl4gkz7LSlORxTuwTbae1eQqNdxdeB+0ilMFCEUc+3NGCNM0RWd+sP5+gzMX
BDQAI1Sc9XaPIg8t3du5JChAl1ifpu/uERZ2WQgtxeBDO6z1Xoa5qz4svf5oURjP
ZjxS0WUKht48Z2rIjk5lZzERSaY3RrX3UtrnZEIzCmInXOrcRPeAD4ZutpiwuHe6
2ABsjuMRnKbATbOUiLdknNyPYYQz2g==
-----END CERTIFICATE-----
```

**Comment by reaperhulk** [comment]
There's (broadly speaking) two ways we can approach this. We can ask that people register OID + decoded shape and then use an ASN.1 parser to handle it, or we can return the extension ASN.1 DER payload without attempting to interpret it and allow the user to do what they want with the data.

The second option is much simpler so I'm leaning that way (at least for now). Do you have a preference?

Incidentally if this extension had been flagged critical cryptography would have raised an error on parse, but since it's not critical we ignore it right now.

**Comment by dainnilsson** [comment]
The second option would cover my needs, so I'm fine with that. Either way you want to go!

**Comment by reaperhulk** [comment]
@dainnilsson is it okay if we use this example certificate in our test suite if/when we choose to implement this?

**Comment by dainnilsson** [comment]
Sure, no problem!

**Comment by alex** [comment]
Fixed by #2607.
